modules/pricer/src/main/java/com/opengamma/strata/pricer/credit/cds/IsdaCompliantDiscountCurveCalibrator.java

```diff
@@ -0,0 +1,454 @@
+   * @return the ISDA compliant discount curve
+   */
+  public IsdaCompliantZeroRateDiscountFactors calibrate(
+      List<IsdaCreditCurveNode> curveNode,
+      LocalDate curveValuationDate,
+      DayCount curveDcc,
+      CurveName name,
+      Currency currency,
+      MarketData marketData,
+      boolean computeJacobian,
+      ReferenceData refData) {
+
+    ArgChecker.isTrue(curveNode.size() > 1, "the number of curve nodes must be greater than 1");
+    LocalDate curveSnapDate = marketData.getValuationDate();
+    int nNode = curveNode.size();
+    BasicFixedLeg[] swapLeg = new BasicFixedLeg[nNode];
+    double[] termDepositYearFraction = new double[nNode];
+    double[] curveNodeTime = new double[nNode];
+    double[] rates = new double[nNode];
+    Builder<ParameterMetadata> paramMetadata = ImmutableList.builder();
+    int nTermDeposit = 0;
+    LocalDate curveSpotDate = null;
+    for (int i = 0; i < nNode; i++) {
+      LocalDate cvDateTmp;
+      IsdaCreditCurveNode node = curveNode.get(i);
+      rates[i] = marketData.getValue(node.getObservableId());
+      LocalDate adjMatDate = node.getNodeDate(curveSnapDate, refData);
+      paramMetadata.add(node.metadata(adjMatDate));
+      if (node instanceof DepositIsdaCreditCurveNode) {
+        DepositIsdaCreditCurveNode termDeposit = (DepositIsdaCreditCurveNode) node;
+        cvDateTmp = termDeposit.getSpotDateOffset().adjust(curveSnapDate, refData);
+        curveNodeTime[i] = curveDcc.relativeYearFraction(cvDateTmp, adjMatDate);
+        termDepositYearFraction[i] = termDeposit.getDayCount().relativeYearFraction(cvDateTmp, adjMatDate);
+        ArgChecker.isTrue(nTermDeposit == i, "TermDepositCurveNode should not be after FixedIborSwapCurveNode");
+        ++nTermDeposit;
+      } else if (node instanceof SwapIsdaCreditCurveNode) {
+        SwapIsdaCreditCurveNode swap = (SwapIsdaCreditCurveNode) node;
+        cvDateTmp = swap.getSpotDateOffset().adjust(curveSnapDate, refData);
+        curveNodeTime[i] = curveDcc.relativeYearFraction(cvDateTmp, adjMatDate);
+        BusinessDayAdjustment busAdj = swap.getBusinessDayAdjustment();
+        swapLeg[i] = new BasicFixedLeg(cvDateTmp, cvDateTmp.plus(swap.getTenor()), swap.getPaymentFrequency().getPeriod(),
+            swap.getDayCount(), curveDcc, busAdj, refData);
+      } else {
+        throw new IllegalArgumentException("unsupported cuve node type");
+      }
+      if (i > 0) {
+        ArgChecker.isTrue(curveNodeTime[i] - curveNodeTime[i - 1] > 0, "curve nodes should be ascending in terms of tenor");
+        ArgChecker.isTrue(cvDateTmp.equals(curveSpotDate), "spot lag should be common for all of the curve nodes");
+      } else {
+        ArgChecker.isTrue(curveNodeTime[i] >= 0d, "the first node should be after curve spot date");
+        curveSpotDate = cvDateTmp;
+      }
+    }
+    ImmutableList<ParameterMetadata> parameterMetadata = paramMetadata.build();
+    CurveMetadata baseMetadata = DefaultCurveMetadata.builder()
```

Comment:
Curves.zeroRates(CurveName, DayCount) is used. 
The parameter metadata list is kept separated for the latter purpose.